pyrobosim/pyrobosim/core/world.py

```diff
@@ -624,19 +624,20 @@ def remove_all_objects(self, restart_numbering=True):
         if restart_numbering:
             self.object_instance_counts = {}
 
-    def update_bounds(self):
+    def update_bounds(self, entity):
         """
         Updates the X and Y bounds of the world.
-
-        TODO: If we're just adding a single room, we only need to check that one
-        and there is probably a more efficient way to do this.
         """
-        for entity in itertools.chain(self.rooms, self.hallways):
+        if isinstance(entity, (Room, Hallway)):
             (xmin, ymin, xmax, ymax) = entity.polygon.bounds
             self.x_bounds[0] = min(self.x_bounds[0], xmin)
             self.x_bounds[1] = max(self.x_bounds[1], xmax)
             self.y_bounds[0] = min(self.y_bounds[0], ymin)
             self.y_bounds[1] = max(self.y_bounds[1], ymax)
```

Comment:
WAIT WAIT WAIT!

If we're *adding* a room/hallway, this is fine... but if we're removing, we need to actually loop through everything. Sorry!

Reply:
You could be more clever for the removing objects case and do something like:

```
if (self.x_bounds[0] == xmin):
    # Loop through all entities because the thing we're removing set the min dimension
# Else do nothing
```

... and repeat for the other 3 bounds. Some helper function might be useful if you go down this route

Reply:
Do we need to actually check bounds for `remove_hallway`,  since the min and max bounds are only affected by adding/ removing of rooms ? and a hallway is always between the bounds of the 2 rooms that it connects

Reply:
And another bug I didn't catch, actually. In the `__init__()`, the bounds are initialized as such:

```
        # World bounds
        self.x_bounds = [0, 0]
        self.y_bounds = [0, 0]
```

Which is a problem if we're just doing `min` / `max`. So here, we'll likely need to initialize the bounds to `None` as well, or an empty list. So then,

* If bounds are None and we're adding an object (the first object), then simply set the bounds directly
* If we removed the last remaining object, the bounds will need to be reset to None

Reply:
Now I remember why I left a TODO in this function, haha :)

Reply:
> Do we need to actually check bounds for `remove_hallway`, since the min and max bounds are only affected by adding/ removing of rooms ? and a hallway is always between the bounds of the 2 rooms that it connects Same case for `add_hallway`

Yes, we do. I know for a fact because I previously had it the way you suggested and ran into a bug with the RRT when trying to do this plan:

![image](https://user-images.githubusercontent.com/4603398/221338971-aa9beca9-58c8-4895-9334-f64e13c19d7a.png)

Reply:
> > Do we need to actually check bounds for `remove_hallway`, since the min and max bounds are only affected by adding/ removing of rooms ? and a hallway is always between the bounds of the 2 rooms that it connects Same case for `add_hallway`
> 
> Yes, we do. I know for a fact because I previously had it the way you suggested and ran into a bug with the RRT when trying to do this plan:
> 
> ![image](https://user-images.githubusercontent.com/4603398/221338971-aa9beca9-58c8-4895-9334-f64e13c19d7a.png)

ohhhhh !
Hallways does not necessarily be a single straight connection !!

Reply:
Yeah, there is an option to specify a "centerline" with arbitrary points, so your hallway could be quite crazy, hehe

Reply:
- Added a `remove` parameter for `update_bounds()` ,
It would not be possible to have `entity` with default `None` since we need the `entity` to check if it sets bounds when removing
- Added `update_bounds()` on `remove_hallways` since it is possible that a hallway was removed on its own.
- Change default `self.x_bounds` and `self.y_bounds` to `None`
- Add checks for when adding first room and deleting last room